src/system-plugins/ui-manager/base.ejs

```diff
@@ -98,23 +100,217 @@ ga('send', 'pageview');
       lazyRegister: true
     };
   </script>
-  <script>window.openrovtheme='<%=theme%>'</script>
+  <script>window.openrovtheme='<%=theme%>'</script>  
+  <link id="lazyScripts" rel="import" href="/all_scripts.html" async>
   <link rel="import" href="components/polymer/polymer.html">
-  <!-- Load the browser global services -->
-  <% for(var i=0; i<scripts.length; i++) {%>
-  <script type="text/javascript" src="<%= scripts[i] %>"></script><% } %>
+<style>
+    body.loading #splash {
+        opacity: 1;
+        z-index: 0;
```

Comment:
I think this is where the black flicker is happening. Changing the z-index here to 9000 (or anything greater than cockpit's z-index) got rid of the flicker